devenv-generate/src/main.rs

```diff
@@ -201,6 +236,19 @@ async fn main() -> Result<()> {
     Ok(())
 }
 
+/// Checks if the telemetry warning has been shown for this project
+fn has_shown_telemetry_warning() -> bool {
+    let dotfile = Path::new(".devenv/telemetry-warning-shown");
```

Comment:
Maybe include `generate` in the name to make it clear it is specific to that feature.